### DISCUSSION TITLE
API: forward to [id].ts if index hit, and no index.ts exists

### DISCUSSION THREAD

**Discussion by JoeRoddy** [vercel/next.js#41132] [Help] [answered]
Hi,  I want to set up a file structure where I handle all my crud logic in the same file`/api/example/[id].ts`.

I also want to be able to hit `/api/example` for POST requests and have them handled by the `[id].ts` file, but that will 404.  Is there any way to get root(`/`) requests sent to `[id].ts` files if no `index.ts` file exists?

I know about this as a way to achieve it:
 https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes
 
 The problem is, now I'm responsible for handling `/api/example/something/doesnt-exist` and throwing a 404 manually, which is not ideal.  I really want to **only** handle the case of forwarding index to /[id].ts ideally.

**Comment by gaston-flores** [comment, marked as answer]
I think you could probably do this with [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites), but it might be easier to create `index.ts` and import your `[id].ts` handler there. So for example:
```js
// /api/users/[id].js
export default function handler(req, res) {
  res.status(200).send(`Hello from ${req.query.id ? req.query.id : "index"}`);
}
```
```js
// /api/users/index.js
import handler from "./[id]";

export default handler;
```
Will handle both requests to `/api/users/` and `/api/users/123`. [Live example](https://codesandbox.io/s/next-id-catch-index-b8vvjd?file=/pages/api/users/index.js).

**Comment by JoeRoddy** [comment]
The import / export trick works great, thanks a ton @gaston-flores !
